common/djangoapps/student/signals/receivers.py

```diff
@@ -17,3 +23,18 @@ def update_last_login(sender, user, **kwargs):  # pylint: disable=unused-argumen
     if not waffle().is_enabled(PREVENT_AUTH_USER_WRITES):
         user.last_login = timezone.now()
         user.save(update_fields=['last_login'])
+
+
+def on_user_updated(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Check for retired usernames.
+    """
+    # Check only at User creation time and when not raw.
+    if not instance.id and not kwargs['raw']:
+        # Check for username that's too close to retired username format.
+        if instance.username.startswith(RETIRED_USERNAME_START):
+            raise AccountValidationError(USERNAME_EXISTS_MSG_FMT.format(username=instance.username), field="username")
```

Comment:
In cases where the format begins with the format token, isn't this check still going to fail?  In that case, RETIRED_USERNAME_START would be the empty string, which every string starts with.

Reply:
Indeed it will! I'm regretting ever checking for similar usernames to the format. I will fix.